website/docs/tutorials/08-interacting-with-starknet/scripting.md

```diff
@@ -0,0 +1,194 @@
+
+### Bash script
+
+Let's assume we want to automate deploying our contract. But before we deploy, we want to first run the tests and check if everything is alright. Instead of manually invoking all the necessary commands, we could write a bash script for that.
+
+#### Setup the script
+
+Let's start with something like this:
+
+```shell
+#!/bin/bash
+
+set -e
+```
+
+The first line informs the system which interpreter should be used to run the script.
+
+The `set -e` instruction tells the interpreter to exit the script immediately if any command returns a non-zero status. This is good for us because we don't want to run the following instructions if one of them fails, for example, we do not want to deploy a contract if the tests for this contract failed.
+
+#### Make sure the contract is correct
+
+```shell
+protostar build
+protostar test
+```
+
+In order to be able to use this, you have to have protostar installed. You can see how to do this [here](../02-installation.md).
+
+These two instructions assure that the contract builds properly and all tests pass.
+
+#### Declare and deploy the contract
+
+Now, we need to first [declare](./03-declare.md) the contract and then [deploy](./04-deploy.md) it.
+
+Normally, we would start with something like this:
+
+```shell
+protostar declare ./build/main.json
+```
+
+But `deploy` command needs the contract's class hash that comes from the `declare`'s command output. Therefore we need to get this output in a standardized way. That's when the `--json` flag comes into play.
+
+By doing:
+
+```shell
+protostar declare ./build/main.json --json
+```
+
+we get an output like this:
+
+```json
+{"class_hash":"0x07a70656b5612a2f87bd98af477c0be5fa2113d13fe1069e55ad326a3e6f4fe6","transaction_hash":"0x01f6a2c391d1bd0a51322ba73037ada20e0b30da8232bb86028f813a0d4c1fdb"} 
+```
+
+Now, we can parse the json and pull all the desired information from it easily as json is a format that is widely supported.
+
+We could do something like this in our bash script:
+
+```shell
+OUTPUT=$(protostar declare ./build/main.json --json)
+CLASS_HASH=$(python -c "import sys, json; print(json.loads(sys.argv[1])['class_hash'])" $OUTPUT)
+protostar deploy $CLASS_HASH --inputs 100
+```
+
+#### Call the contract
+
+Now, let's say we want to call our contract. In this case, we need the contract address that is being returned from the `deploy` command call.
+
+We are basically going to do the same thing as previously to pass the contract address from `deploy` to `call`:
+
+```shell
+OUT=$(protostar deploy $CLASS_HASH --inputs 100 --json)
+CONTRACT_ADDRESS=$(python -c "import sys, json; print(json.loads(sys.argv[1])['contract_address'])" $OUT)
+protostar call --contract-address $CONTRACT_ADDRESS --function get_balance
+```
+
+In the end, our bash script looks like this:
+
+```shell
```

Comment:
I would write an example script in python i believe it will be more useful for users

Reply:
I think the fact you can use bash is pretty obvious for somebody that knows bash. 

> Most of the machines do have bash

Python as well. I would prefer to direct user towards python as it might be not obvious to the user that it is an option. We don't want a steep learning curve for protostar.

Reply:
Discussed on slack, we should use python as it is going to be simpler to read and reuse